src/components/Sidebar.tsx

```diff
@@ -292,7 +292,9 @@ function DropdownMenu({ title, icon, menuItems, onClose } : {
             // Only sets the link it is a url 
             {...isUrl && { as: NextLink,  href: item.action } }
             onClick={() => {
-              (item as { action: Function }).action();
+              // Ensure type checking before calling the function.
+              // Otherwise it will cause run time error if action is a URL.
```

Comment:
With the new code the logic is obvious and self descriptive. the comment is no longer needed. it is outdated anyway.